db/view/view.cc

```diff
@@ -1546,19 +1546,23 @@ bool needs_static_row(const mutation_partition& mp, const std::vector<view_and_b
 // If the assumption that the given base token belongs to this replica
 // does not hold, we return an empty optional.
 static std::optional<gms::inet_address>
-get_view_natural_endpoint(const locator::effective_replication_map_ptr& erm, bool network_topology,
-        const dht::token& base_token, const dht::token& view_token) {
-    auto& topology = erm->get_token_metadata_ptr()->get_topology();
+get_view_natural_endpoint(
+        const locator::effective_replication_map_ptr& base_erm,
+        const locator::effective_replication_map_ptr& view_erm,
```

Comment:
One should either pass a _ptr by value, or dereference it and pass the pointee. The double reference is just a slowdown.

It's an existing issue, so no need to fix.